include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -102,32 +102,28 @@ struct __transform_fn
     template<typename _ExecutionPolicy, std::ranges::random_access_range _R1, std::ranges::random_access_range _R2,
              std::ranges::random_access_range _OutRange, std::copy_constructible _F, typename _Proj1 = std::identity,
              typename _Proj2 = std::identity>
-    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && (std::ranges::sized_range<_R1>
-        || std::ranges::sized_range<_R2>) && std::ranges::sized_range<_OutRange>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R1>
+        && std::ranges::sized_range<_R2> && std::ranges::sized_range<_OutRange>
         && std::indirectly_writable<std::ranges::iterator_t<_OutRange>,
-            std::indirect_result_t<_F&, std::projected<std::ranges::iterator_t<_R1>, _Proj1>,
-            std::projected<std::ranges::iterator_t<_R2>, _Proj2>>>
+        std::indirect_result_t<_F&, std::projected<std::ranges::iterator_t<_R1>, _Proj1>,
+        std::projected<std::ranges::iterator_t<_R2>, _Proj2>>>
 
     std::ranges::binary_transform_result<std::ranges::borrowed_iterator_t<_R1>, std::ranges::borrowed_iterator_t<_R2>,
-        std::ranges::borrowed_iterator_t<_OutRange>>
+                                         std::ranges::borrowed_iterator_t<_OutRange>>
     operator()(_ExecutionPolicy&& __exec, _R1&& __r1, _R2&& __r2, _OutRange&& __out_r, _F __binary_op,
                _Proj1 __proj1 = {}, _Proj2 __proj2 = {}) const
     {
         const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
 
         using _Size = std::common_type_t<oneapi::dpl::__internal::__range_size_t<_R1>,
             oneapi::dpl::__internal::__range_size_t<_R2>, std::ranges::range_size_t<_OutRange>>;
-        _Size __size = std::ranges::size(__out_r);
-        if constexpr(std::ranges::sized_range<_R1>)
-            __size = std::ranges::min(__size, (_Size)std::ranges::size(__r1));
-        if constexpr(std::ranges::sized_range<_R2>)
-            __size = std::ranges::min(__size, (_Size)std::ranges::size(__r2));
+        const _Size __size = std::ranges::min({(_Size)std::ranges::size(__r1), (_Size)std::ranges::size(__r2),
+            (_Size)std::ranges::size(__out_r)});
```

Comment:
Could you make the same adjustment in `std_ranges_transform.pass.cpp`? https://github.com/uxlfoundation/oneDPL/blob/f49feb459eb0fc19524a2cd0b5789e5b96575b5e/test/parallel_api/ranges/std_ranges_transform.pass.cpp#L51-L57

Could you also use `take_view` instead of a subrange in `transform_binary_checker` like it is done below in the library?

Reply:
done